dz1-4.js

```diff
@@ -0,0 +1,34 @@
+function getRequest() {
+		if ( window.XMLHttpRequest) {
+			return new XMLHttpRequest();
+		} else if ( window.ActiveXObject) {
+			return new ActiveXObject("Microsoft.XMLHTTP");
+		}
+}
+
+function makeGETRequest (url) {
+		return new Promise ((resolve, reject) => {
+				
+		const xhr = this.getRequest();
+		
+		xhr.onreadystatechange = function () {
+			if (xhr.readyState !== 4) return;
+			if (xhr.status === 200) { 
+				try { resolve (JSON.parse(xhr.responseText));
+				} catch(e) { reject(e.message); }
+			} else reject(xhr.status);
+		}
+				
+		xhr.open('GET', url, true);
```

Comment:
```suggestion
		xhr.open('GET', url);
```
тут по умолчанию прописывается `true`